@xen-orchestra/web-core/lib/components/ui/table-pagination/UiTablePagination.vue

```diff
@@ -0,0 +1,171 @@
+<!-- v2 -->
+<template>
+  <div class="ui-table-pagination">
+    <div class="buttons-container">
+      <PaginationButton :disabled="isFirstPage" :icon="faAngleDoubleLeft" @click="goToFirstPage()" />
+      <PaginationButton :disabled="isFirstPage" :icon="faAngleLeft" @click="goToPreviousPage()" />
+      <PaginationButton :disabled="isLastPage" :icon="faAngleRight" @click="goToNextPage()" />
+      <PaginationButton :disabled="isLastPage" :icon="faAngleDoubleRight" @click="goToLastPage()" />
+    </div>
+    <span class="typo p3-regular label">
+      {{ $t('core.select.n-object-of', { from: startIndex, to: endIndex, total: totalItems }) }}
+    </span>
+    <span class="typo p3-regular label show">{{ $t('core.show-by') }}</span>
+    <div class="dropdown-wrapper">
+      <select v-model="pageSize" :disabled class="dropdown typo c3-regular" @change="goToFirstPage">
+        <option v-for="option in pageSizeOptions" :key="option" :value="option" class="typo p2-medium">
+          {{ option }}
+        </option>
+      </select>
+      <VtsIcon class="icon" accent="current" :icon="faAngleDown" />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import PaginationButton from '@core/components/ui/table-pagination/PaginationButton.vue'
+import {
+  faAngleDoubleLeft,
+  faAngleDoubleRight,
+  faAngleDown,
+  faAngleLeft,
+  faAngleRight,
+} from '@fortawesome/free-solid-svg-icons'
+import { useOffsetPagination } from '@vueuse/core'
+import { computed, ref, watch } from 'vue'
+
+const { totalItems } = defineProps<{
+  totalItems: number
+  disabled?: boolean
+}>()
+
+const emit = defineEmits<{
+  change: [{ currentPage: number; pageSize: number; startIndex: number; endIndex: number }]
+}>()
```

Comment:
I'd suggest extracting typing here.

```suggestion
export type PaginationPayload = {
  currentPage: number
  pageSize: number
  startIndex: number
  endIndex: number
}

const { totalItems } = defineProps<{
  totalItems: number
  disabled?: boolean
}>()

const emit = defineEmits<{
  change: [payload: PaginationPayload]
}>()
```